### PR TITLE
V4 documentation

### PR DESCRIPTION
**Pull request by DanteRangelR**
Add GET    /merchant
Add POST  /bills
Add GET     /bills
Add GET     /bills/{id}
Add GET     /bills/{id}/challenges
Add POST   /bill
Add POST   /migration
Add POST   /migration_challenge
Add PUT      /challenges/{id}
Add Bill Structure

### CODE REVIEW

File: apix_documentation_swagger_v4.yaml [hunk 1]
```diff
@@ -685,7 +699,394 @@ paths:
                         parameter: id
                       code: NotFound
                       detail: The resource requested can't be found
+  /merchant:
+    post:
```
[the remark this answers is not in the document]
Comment (DanteRangelR) [reply apix_documentation_swagger_v4.yaml]: In each one of the CRUD, you can get to the element through of post request adding a body with id and your value

File: apix_documentation_swagger_v4.yaml [hunk 2]
```diff
@@ -685,7 +699,394 @@ paths:
...
+      tags:
+        - xData
+      summary: Retrieves a specific bill’s cached data
+      operationId: get specify Bill
+      description: |
+        This endpoint retrieves a specific bill’s cached data.
+      parameters:
+        - name: id
+          in: path
+          description: UUID of the bill you wish to obtain (this version still
```
[the remark this answers is not in the document]
Comment (jdsampayo) [reply apix_documentation_swagger_v4.yaml]: Now id -> uuid, so, it is fine to name them ID, what happens now is that we don't have anymore the pair legacy_id / uuid. But we can name an UUID just ID.

File: apix_documentation_swagger_v4.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (jdsampayo) [reply apix_documentation_swagger_v4.yaml]: Just remove this one of /merchant

File: apix_documentation_swagger_v4.yaml [hunk 3]
```diff
@@ -685,7 +699,394 @@ paths:
                         parameter: id
                       code: NotFound
                       detail: The resource requested can't be found
+  /merchant:
+    post:
+      tags:
+        - Common
+      summary: Obtains the details of a single merchant
+      operationId: getMerchantPost
```
[the remark this answers is not in the document]
Comment (jdsampayo) [reply apix_documentation_swagger_v4.yaml]: Just not the merchant one, the bills is ok

File: apix_documentation_swagger_v4.yaml [hunk 4]
```diff
@@ -685,7 +699,394 @@ paths:
...
+                  example: 6ab564f9-4e9d-4b03-96d6-dd9f2c39ec53
+                login:
+                  type: string
+                  description: User name 
+                  example: email@exaple.com
+                password:
+                  type: string
+                  description: Password
+                  example: Example123!
+                external_id _dynamic:
```
[the remark this answers is not in the document]
Comment (jdsampayo) [reply apix_documentation_swagger_v4.yaml]: external_id _dynamic -> external_id_dynamic

File: apix_documentation_swagger_v4.yaml [hunk 5]
```diff
@@ -685,7 +699,394 @@ paths:
...
+                  type: string
+                  description: In merchant configuration you will find the custom fields external id
```
[the remark this answers is not in the document]
Comment (jdsampayo) [reply apix_documentation_swagger_v4.yaml]: I don't think the customers can see the Configuration, what does the description means?

**Pull request merged**